### PR TITLE
refactor: embed component metadata

### PR DESCRIPTION
**Pull request by vados-cosmonic**
## Feature or Problem

This refactors the existing branch implementing Go WASI p2 components with code to embed component metadata.

## Related Issues

https://github.com/wasmCloud/wash/issues/512

## Consumer Impact

Consumers will be able to build WASI preview2 components from Go with `wash build`

## Testing

Built on platform(s)
- [x] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

Tested on platform(s)
- [x] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

### Manual Verification

Verified manually locally w/ the [wasmcon golang codebase](https://github.com/cosmonic/wasmcon-workshop/tree/main/hello-cosmo-go)

### CODE REVIEW

File: crates/wash-lib/src/build.rs [hunk 1]
```diff
@@ -95,14 +97,21 @@ pub fn build_actor(
...
+                .as_ref()
+                .context("`wit_world` must be specified in wasmcloud.toml for creating preview1 or preview2 components")?,
+            &actor_wasm_path,
+            &actor_wasm_path,
+        )?;
+    };
 
-    // If the actor has been configured as WASI Preview2, adapt it
-    if actor_config.wasm_target == WasmTarget::WasiPreview2
-        || actor_config.wasm_target == WasmTarget::TinyGoWasiPreview1
```
Comment (ricochet) [inline crates/wash-lib/src/build.rs:104]: So essentially if it has not been configured for Preview2 we do not adapt. This avoids the situation @jordan-rash pointed out where you need to target wasi to build wasm-only (no js glue) tiny-go modules.

File: crates/wash-lib/src/build.rs [hunk 2]
```diff
@@ -330,6 +339,76 @@ pub(crate) fn get_wasi_preview2_adapter_bytes(config: &ActorConfig) -> Result<Ve
     Ok(wasmcloud_component_adapters::WASI_PREVIEW1_REACTOR_COMPONENT_ADAPTER.into())
 }
 
+/// Embed the WIT inside a given WASM binary, in-place.
```
Comment (ricochet) [inline crates/wash-lib/src/build.rs]: OK this is the nittiest of nits:
https://www.w3.org/TR/wasm-core/intro/introduction.html#wasm

A contraction of “WebAssembly”, not an acronym, hence not using all-caps.

File: crates/wash-lib/src/build.rs (hunk 1 above)
Comment (vados-cosmonic) [reply crates/wash-lib/src/build.rs:104]: Yup, that's absolutely right -- so right now, while we embed component metadata whether you target p1 or p2, we only *adapt* if you target p2 (to take you from p1 to p2).

The `wasmcloud.toml` in the wasmcon demo when modified looks like this:

```toml
name = "hello-cosmo-go"
language = "tinygo"
type = "actor"
version = "0.1.0"

[actor]
claims = ["wasmcloud:httpserver", "wasmcloud:keyvalue", "wasmcloud:builtin:logging"]
wasm_target = "wasm32-wasi-preview2"
wit_world = "hello-cosmo"
```

**Review by ricochet — APPROVED** [review]

File: crates/wash-lib/src/build.rs (hunk 2 above)
Comment (vados-cosmonic) [reply crates/wash-lib/src/build.rs]: Absolutely will fix, thank you, thanks for spotting this

- [x] remove WASM usage in favor of WebAssembly

File: crates/wash-lib/src/build.rs [hunk 3]
```diff
@@ -330,6 +339,72 @@ pub(crate) fn get_wasi_preview2_adapter_bytes(config: &ActorConfig) -> Result<Ve
...
+            wit_dir.display()
+        );
+    };
+
+    // Resolve the WIT directory packages & worlds
+    let mut resolve = wit_parser::Resolve::default();
+    let (package_id, _paths) = resolve
+        .push_dir(&wit_dir)
+        .context("failed to add WIT deps directory")?;
+    log::info!("successfully loaded WIT @ [{}]", wit_dir.display());
```
Comment (connorsmith256) [inline crates/wash-lib/src/build.rs:363]: Nit: debug level? I don't think users need to see this every time they `wash build`

(Same with the other logs in this function)

**Review by connorsmith256 — CHANGES_REQUESTED** [review]

File: crates/wash-lib/src/build.rs (hunk 3 above)
Comment (vados-cosmonic) [reply crates/wash-lib/src/build.rs:363]: Ah so the default level is *above* info -- it's error (i.e. someone has to do `RUST_LOG=info` to see these). 

Also, I think info is the right level for these messages -- they don't touch on the really deep "how", and they mostly say when something happened successfully/happy path. 

In my mind debug messages generally cover the before, during and end of some operation (so we can debug) where as informational stuff just lets you know general outcomes from time to time.

Comment (connorsmith256) [reply crates/wash-lib/src/build.rs:363]: I don't think I realized until this moment that info wasn't the default level 😮 my bad!

**Pull request merged**